integration/app/app.component.spec.ts

```diff
@@ -220,4 +220,61 @@ describe('until-destroy runtime behavior', () => {
     fixture.destroy();
     expect(service.disposed).toBeTruthy();
   });
+
+  it('should be able to re-use methods of the singleton service multiple times', () => {
+    // Arrange
+    let disposedTimes = 0;
+    let startCalledTimes = 0;
+    let originalStopCalledTimes = 0;
+
+    @Injectable()
+    class IssueSixtySixService {
+      start(): void {
+        startCalledTimes++;
+
+        new Subject()
+          .pipe(
+            untilDestroyed(this, 'stop'),
```

Comment:
I just notice that we're using a `string` for the method type. We can make it safer by using the `keyof` type.

Reply:
Good idea. Should it be in a separate PR?

Reply:
No, let's add it here. Thanks!

Reply:
@NetanelBasal ping

Reply:
💯